apt/utils.py

```diff
@@ -29,6 +29,134 @@ def get_iris_dataset():
+        train = data.iloc[train_set]
+        test = data.iloc[test_set]
+    x_train = train.drop(["label"], axis=1)
+    y_train = train.loc[:, "label"]
+    x_test = test.drop(["label"], axis=1)
+    y_test = test.loc[:, "label"]
+
+    categorical_features = ["Existing_checking_account", "Credit_history", "Purpose", "Savings_account",
+                            "Present_employment_since", "Personal_status_sex", "debtors", "Property",
+                            "Other_installment_plans", "Housing", "Job"]
+    x_train.reset_index(drop=True, inplace=True)
+    y_train.reset_index(drop=True, inplace=True)
+    x_test.reset_index(drop=True, inplace=True)
+    y_test.reset_index(drop=True, inplace=True)
+
+    return (x_train, y_train), (x_test, y_test)
+
+
+def _modify_german_dataset(data):
+
+    def modify_Foreign_worker(value):
+        if value == 'A201':
+            return 1
+        elif value == 'A202':
+            return 0
+        else:
+            raise Exception('Bad value')
+
+    def modify_Telephone(value):
+        if value == 'A191':
+            return 0
+        elif value == 'A192':
+            return 1
+        else:
+            raise Exception('Bad value')
+
+    def modify_Existing_checking_account(value):
+        if value == 'A11':
+            return 'under_zero'
+        elif value == 'A12':
+            return 'zero_to_200'
+        elif value == 'A13':
+            return 'above_200'
+        elif value == 'A14':
+            return 'no_checking_account'
+        else:
+            raise Exception('Bad value')
+
+    def modify_Credit_history(value):
+        if value == 'A30':
+            return 'no credits taken'
+        elif value == 'A31':
+            return 'all credits paid back duly'
+        elif value == 'A32':
+            return 'existing credits paid back duly till now'
+        elif value == 'A33':
+            return 'delay in paying off in the past'
+        elif value == 'A34':
+            return 'critical account'
+        else:
+            raise Exception('Bad value')
+
+
+    def modify_Purpose(value):
+        if value == 'A40':
+            return 'new car'
+        elif value == 'A41':
+            return 'used car'
+        elif value == 'A42':
+            return 'furniture/equipment'
+        elif value == 'A43':
+            return 'radio/television'
+        elif value == 'A44':
+            return 'domestic appliances'
+        if value == 'A45':
+            return 'repairs'
+        elif value == 'A46':
+            return 'education'
+        elif value == 'A47':
+            return 'vacation'
+        elif value == 'A48':
+            return 'retraining'
+        elif value == 'A49':
+            return 'business'
+        elif value == 'A410':
+            return 'other'
+        else:
+            raise Exception('Bad value')
+
+    # data['Existing_checking_account'] = data['Existing_checking_account'].apply(modify_Existing_checking_account)
```

Comment:
Why are these commented out?

Reply:
because they still categorical features, so I did not apply this to these features
I did this just to the features that can be moved to numerical features (if it has two values)

Reply:
Ok, so please remove all code that is not being used.